src/lib/operations.ts

```diff
@@ -121,6 +121,124 @@ const prepareToEditExtrude: PrepareToEditCallback =
     }
   }
 
+/**
+ * Gather up the argument values for the Chamfer or Fillet command
+ * to be used in the command bar edit flow.
+ */
+const prepareToEditEdgeTreatment: PrepareToEditCallback = async ({
+  operation,
+  artifact,
+}) => {
+  const isChamfer =
+    artifact?.type === 'edgeCut' && artifact.subType === 'chamfer'
+  const isFillet = artifact?.type === 'edgeCut' && artifact.subType === 'fillet'
+  const baseCommand = {
+    name: isChamfer ? 'Chamfer' : 'Fillet',
+    groupId: 'modeling',
+  }
+  if (
+    operation.type !== 'StdLibCall' ||
+    !operation.labeledArgs ||
+    (!isChamfer && !isFillet)
+  ) {
+    return baseCommand
+  }
+
+  // Recreate the selection argument (artiface and codeRef) from what we have
+  const edgeArtifact = getArtifactOfTypes(
+    {
+      key: artifact.consumedEdgeId,
+      types: ['segment', 'sweepEdge'],
+    },
+    engineCommandManager.artifactGraph
+  )
+  if (err(edgeArtifact)) {
+    return baseCommand
+  }
```

Comment:
What happens here exactly when there's an error and it returns the baseCommand?

Does the edit still work?

Reply:
Yeah you're right @pierremtb there are probably places in `operations.ts` that you took from that are also not using that pattern yet, apologies it was a pattern added while we've been building the system but we should report a failure reason everywhere instead of returning `baseCommand` silently.